RecoMTD/TrackExtender/plugins/TrackExtenderWithMTD.cc

```diff
@@ -267,21 +374,78 @@ void TrackExtenderWithMTDT<TrackCollection>::produce(edm::Event& ev, const edm::
   ev.getByToken(bsToken_, bsH);
   const auto& bs = *bsH;
 
+  const Vertex* pv = nullptr;
+  if (useVertex_ && !useSimVertex_) {
+    edm::Handle<VertexCollection> vtxH;
+    ev.getByToken(vtxToken_, vtxH);
+    if (vtxH.isValid())
+      pv = &(vtxH.product()->at(0));
+  }
+
+  const SimVertex* genPV = nullptr;
+  if (useVertex_ && useSimVertex_) {
+    edm::Handle<vector<SimVertex>> genVtxH;
+    ev.getByToken(genVtxToken_, genVtxH);
+    if (genVtxH.isValid())
+      genPV = &(genVtxH.product()->at(0));
+  }
+
+  double vtxTime = 0.;
+  if (useVertex_) {
+    if (useSimVertex_ && genPV) {
+      vtxTime = genPV->position().t() * CLHEP::second;  //convert to ns
```

Comment:
@cvuosalo can you recommend a better unit conversion here?

Reply:
Yes, please remove

`#include "CLHEP/Units/GlobalSystemOfUnits.h"`

and add

`#include DataFormats/Math/interface/GeantUnits.h`

I will provide the suggested change for line 396.